src/interface/cmdTlmServer.c

```diff
@@ -0,0 +1,487 @@
+                                         char *pName) {
+    struct mq_attr attr;
+    attr.mq_flags = O_NONBLOCK;
+    attr.mq_maxmsg = maxNumMsg;
+    attr.mq_msgsize = sizeMsg;
+    return mq_open(pName, O_CREAT | O_RDWR | O_NONBLOCK, 0, &attr);
+}
+
+// Prepare the message queues with an identifier used to differentiate
+// different queue messages. User also needs to provide the maximum number of
+// element in telemetry queue.
+// Return 0 if success, otherwise, return -1.
+static int cmdTlmServer_prepareMsgQueue(serverInfo_t *pServerInfo,
+                                        int identifier, long maxNumQueueTlm) {
+    // Generate the names of message queues
+    pServerInfo->pQueueNameCmdStatus =
+        cmdTlmServer_genNameMsgQueue("/queueCmdStatus", identifier);
+    pServerInfo->pQueueNameTlm =
+        cmdTlmServer_genNameMsgQueue("/queueTlm", identifier);
+
+    // Message queue of comomand status
+
+    // Note that the value of 4 is a random number. I expect there will be only
+    // 1 in the real running.
+    // The value here is just by try-and-error. If this value is too big, the
+    // kernel will reject it for the memory allocation.
+    // It is noted that the kernel will restrict the allocated memory for the
+    // message queue. Therefore, this value can not be too big.
+    pServerInfo->msgQueueCmdStatus =
+        cmdTlmServer_createMsgQueue(4, (long)sizeof(commandStatusStructure_t),
+                                    pServerInfo->pQueueNameCmdStatus);
+    if (pServerInfo->msgQueueCmdStatus == (mqd_t)(-1)) {
+        syslog(LOG_ERR, "Failed to create the message queue of command status "
+                        "in %s server.",
+               pServerInfo->pName);
+        return -1;
+    }
+
+    // Allocate the memory of message queue of command status
+    pServerInfo->pMsgCmdStatus =
+        g_malloc0_n((gsize)sizeof(commandStatusStructure_t), 2);
+
+    // Message queue of telemetry
+    pServerInfo->msgQueueTlm = cmdTlmServer_createMsgQueue(
+        maxNumQueueTlm, pServerInfo->sizeMsgTlm, pServerInfo->pQueueNameTlm);
+    if (pServerInfo->msgQueueTlm == (mqd_t)(-1)) {
+        syslog(LOG_ERR,
+               "Failed to create the message queue of telemetry in %s server.",
+               pServerInfo->pName);
+        return -1;
+    }
+
+    // Allocate the memory of message queue of telemetry
+    pServerInfo->pMsgTlm = g_malloc0_n((gsize)pServerInfo->sizeMsgTlm, 2);
+
+    return 0;
+}
+
+// Receive the message with a timeout value (in millisecond). The received
+// message will be written to the memory pointed to by pCmdMsg.
+// Return the error status or the received number of byte.
+// If the return value < 0, it means the timeout from poll() or error from
+// poll() or recv().
+// Note. If we do not set the timeout when using the recv(), the server may not
+// be able to close this thread because the recv() is a blocking call.
```

Comment:
Please document what a value of 0 means. Based on how you use the code below it apparently means that the client closed the connection?